File: c++ version/1-100/66-plus-one.cpp
vector<int> plusOne(vector<int>& digits) {
    list<int> ans;
    int n = digits.size();
    ++digits[n-1];
    int carry = 0;
    int val = 0;
    for (int i = n-1; i >= 0; --i) {
        val = (digits[i] + carry) % 10;
        carry = (digits[i] + carry) > 9 ? 1 : 0;
        ans.insert(ans.begin(), val);
    }
    if (carry == 1) ans.insert(ans.begin(), 1);
    return vector<int>(ans.begin(), ans.end());
}

File: python version/66-plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        digits = digits[::-1]
        digits[0] += 1
        carry = 0
        for i in range(len(digits)):
            sum = digits[i] + carry
            carry, digits[i] = sum//10, sum%10
        if carry:
            digits.append(1)
        return digits[::-1]